src/phpDocumentor/Renderer/Action/AppendFileHandler.php

```diff
@@ -0,0 +1,103 @@
+<?php
+/**
+ * phpDocumentor
+ *
+ * PHP Version 5.4
+ *
+ * @copyright 2010-2014 Mike van Riel / Naenius (http://www.naenius.com)
+ * @license   http://www.opensource.org/licenses/mit-license.php MIT
+ * @link      http://phpdoc.org
+ */
+
+namespace phpDocumentor\Renderer\Action;
+
+use League\Flysystem\Adapter\Local;
+use phpDocumentor\Renderer\Action;
+use phpDocumentor\Renderer\ActionHandler;
+use Symfony\Component\Filesystem\Filesystem;
+use Webmozart\Assert\Assert;
+
+class AppendFileHandler implements ActionHandler
+{
+    /**
+     * @var Filesystem
+     */
+    private $filesystem;
+
+    public function __construct(Filesystem $filesystem)
+    {
+        $this->filesystem = $filesystem;
+    }
+
+    /**
+     * Executes the activities that this Action represents.
+     *
+     * @param AppendFile $action
+     *
+     * @todo this method still acts as a Copy because I currently do not know how to append with stream,
+     *       fix this as soon as I got internet.
+     *
+     * @return void
+     */
+    public function __invoke(Action $action)
+    {
+        $source      = $this->getSourceLocation($action);
+        $destination = $this->getDestination($action);
+
+        Assert::fileExists($source);
+        Assert::writable(dirname($destination));
+
+        $destinationFS = $action->getRenderPass()->getFilesystem();
+
+        if (is_file($source)) {
+            $stream = fopen($source, 'r+');
+            $destinationFS->putStream($destination, $stream);
+            fclose($stream);
+        } else {
+            $sourceFS = new \League\Flysystem\Filesystem(new Local($source));
+            foreach ($sourceFS->listContents('', true) as $path) {
+                $path = $path['path'];
+
+                if ($sourceFS->getMetadata($path)['type'] === 'dir') {
+                    continue;
+                }
+
+                $destinationFS->putStream($destination . '/' . $path, $sourceFS->readStream($path));
+            }
+        }
+    }
+
+    /**
+     * @param AppendFile $action
+     *
+     * @return string
+     */
+    private function getSourceLocation(Action $action)
+    {
+        $source = (string)$action->getSource();
+        if (! $this->filesystem->isAbsolutePath($source)) {
```

Comment:
I have looked into Puli and didn't see any convincing arguments to use it; if anything it would make things more complex as I see it. Perhaps I am wrong; I haven't actually worked with Puli yet but I wasn't convinced for this use-case